### DISCUSSION TITLE
How should we implement Jetpack Compose on the Compose sample folder?

### DISCUSSION THREAD

**Discussion by chornerman** [nimblehq/android-templates#279] [RFCs]
We currently have a plan to create a sample folder to demonstrate Jetpack Compose. The latest approach we use in Android Template is to integrate Jetpack Compose inside the Fragment and still use the traditional XML navigation library for moving from one screen to another, however, we can opt out entirely of the traditional Fragment structure and instead use [the navigation library for Compose](https://developer.android.com/jetpack/compose/navigation), which I believe should be the future of Android development 😎.

Since we have never officially used the new approach (no Fragment involved) on Android Template before, we might not have the complete structure for it yet. This RFC aims to vote on whether it sounds okay enough to implement this approach on the Compose sample folder now, or we should still use the current approach (using Fragment) at the moment.

Here're some pros and cons of using the no Fragment approach

**Advantage?**
- It seems to be the future of Android development and we might have to adapt to this approach soon

**Disadvantage?**
- We don't have the complete structure for it yet, so it might be too soon to implement it directly to the sample folder in Android Template

**Comment by toby-thanathip** [comment]
@nimblehq/android-chapter I voted for "**Integrate it inside Fragment using the current structure we have**" 👍 

Our `template` is currently not adapted to use the new navigation library for Compose. 

If a developer generates a new project, he/she will currently see `nav_graph_main.xml`. 

This will lead to a bigger inconsistency between `template` & `sample-compose`

Being up-to-date with the newest tools sounds great, but I think we should only add it **_later_** once we have proper support for the script to [generate a new project with XML or  Compose](https://github.com/nimblehq/android-templates/issues/214).
 
As @chornerman mentioned, I believe it's simply **_too soon_** (to do it as part of this ticket: #274)

**Reply by lydiasama** [reply]
@Tuubz That's a really good point. 👍🏻

**Reply by lydiasama** [reply]
@Tuubz I changed my vote following you. 😂

**Comment by kaungkhantsoe** [comment]
Did you mean like `Single Activity` approach @chornerman ?

**Reply by chornerman** [reply]
Correct @kaungkhantsoe

**Comment by Wadeewee** [comment]
@Tuubz @chornerman 🙋‍♀️  I voted for `Go with no Fragment approach and using navigation library for Compose` as it's `compose` sample folder, and we suppose to show how to use it, So I think we shouldn't follow the old implementation, but at the same time, I understand it out of scope for this issue [#242](https://github.com/nimblehq/android-templates/issues/242) because we haven't tried to implement it in `template` before. What about if we create an issue for `using navigation library for Compose` and target it in `sample-compose` after we finished [#242](https://github.com/nimblehq/android-templates/issues/242) 🙏

**Reply by toby-thanathip** [reply]
Agree with @Wadeewee, I'll create an issue to `use navigation library for Compose` so we can handle it later 🚀

**Reply by hoangnguyen92dn** [reply]
Agree. We should consider using the single activity architecture with Compose components rather than attach inside Fragment. 🚀

**Reply by lydiasama** [reply]
Voted for this comment! 👍🏻
